cryptographic_estimators/SDEstimator/SDAlgorithms/bjmm_dw.py

```diff
@@ -19,56 +19,61 @@
+    _mitm_nn_complexity,
+    binom,
+    log2,
+    ceil,
+    inf,
+)
 from scipy.special import binom as binom_sp
 from scipy.optimize import fsolve
 from warnings import filterwarnings
 from types import SimpleNamespace
 from ..sd_constants import *
+
 filterwarnings("ignore", category=RuntimeWarning)
 
 
 class BJMMdw(SDAlgorithm):
     def __init__(self, problem: SDProblem, **kwargs):
-        """
-        Construct an instance of BJMM's estimator using *d*isjoint *w*eight distributions combined with
-         MitM-nearest neighbor search. [EB22]_, [MMT11]_, [BJMM12]_.
+        """Construct an instance of BJMM's estimator using disjoint weight distributions combined with MitM-nearest neighbor search.
 
-        Expected weight distribution::
+        Expected weight distribution:
 
             +---------------------------+-------------+------------+----------+----------+----------+----------+
             |<-+ n - k - 2 l1 - 2 l2 +->|<-+ k / 2 +->|<-+ k / 2 ->|<-+ l1 +->|<-+ l1 +->|<-+ l2 +->|<-+ l2 +->|
             |   w - 2 p - 2 w1 - 2 w2   |      p      |      p     |    w1    |    w1    |    w2    |    w2    |
             +---------------------------+-------------+------------+----------+----------+----------+----------+
 
-
-        INPUT:
-
-        - ``problem`` -- syndrome decoding problem instance
-        - ``k`` -- dimension of the code
-        - ``w`` -- Hamming weight of error vector
-        - ``mem`` -- upper bound on the available memory (as log2), default unlimited
-        - ``hmap`` -- indicates if hashmap is being used (default: true)
-        - ``p_range`` -- interval in which the parameter p is searched (default: [0, 25], helps speeding up computation)
-        - ``memory_access`` -- specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage)
-
-        EXAMPLES::
-
-            sage: from cryptographic_estimators.SDEstimator.SDAlgorithms import BJMMdw
-            sage: from cryptographic_estimators.SDEstimator import SDProblem
-            sage: BJMMdw(SDProblem(n=100,k=50,w=10))
+        Args:
+            problem (SDProblem): Syndrome decoding problem instance.
+            k (int): Dimension of the code.
+            w (int): Hamming weight of error vector.
+            mem (int, optional): Upper bound on the available memory (as log2), default unlimited.
+            hmap (bool, optional): Indicates if hashmap is being used (default: True).
+            p_range (list, optional): Interval in which the parameter p is searched (default: [0, 25], helps speeding up computation).
+            memory_access (int, optional): Specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage).
+
+        References:
```

Comment:
ok here we change the "citation type" from `[MMT]` to this. Is there a reason for this? Because I think this way leads to 
1 ) alot of copy paste
2) errors because of that

I liked the first central way of collecting citations, but maybe there are reasons against it.

Reply:
I have no clue why that could happen. I mean, the information to create that reference section doesn't even exist in the previous docstring.

AI was making some really weird things here.